### PR TITLE
Swapna_Assignment_17_40_min

### PR DESCRIPTION
**Pull request by SwapnaWJun20**
image
image

### CODE REVIEW

File: src/swapnaWJun20/CountVariations.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+			if(Character.isDigit(ch)) 
+				dCount++;
+			else if (Character.isLetter(ch)) 
+				letterCount++;
+			else
+				specialCount++;
+
+			if(Character.isLowerCase(ch)) 
+				lCount++;
+			else if(Character.isUpperCase(ch))
```
Comment (KrishnaKTechnocredits) [inline src/swapnaWJun20/CountVariations.java:22]: No two separate if else block is not required. You can achieve output without using letterCount varibale and else if (Character.isLetter(ch))  block. Think !!!

File: src/swapnaWJun20/FirstLastIndex.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	void findIndex(String input)
+	{
+		System.out.println("Input : "+input);
+		char ch =' ';
+
+		for(int index=0;index<input.length();index++)
+		{
+			ch = input.charAt(index);
+
+			if(index == input.indexOf(ch) && input.indexOf(ch) < input.lastIndexOf(ch))
```
Comment (KrishnaKTechnocredits) [inline src/swapnaWJun20/FirstLastIndex.java:14]: Perfect logic.

File: src/swapnaWJun20/CountVariations.java (hunk 1 above)
Comment (SwapnaWJun20) [reply src/swapnaWJun20/CountVariations.java:22]: Updated changes done!!

File: src/swapnaWJun20/CountVariations.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			else if(Character.isLowerCase(ch)) 
+				lCount++;
+			else if(Character.isUpperCase(ch))
+				uCount++;
+			else
+				specialCount++;
+		}
+
+		System.out.println("\nTotal Digits: "+ dCount +
+				"\nTotal Letters: " + (uCount+lCount)+
```
Comment (KrishnaKTechnocredits) [inline src/swapnaWJun20/CountVariations.java:29]: Good job Dear.

**Review by KrishnaKTechnocredits — APPROVED** [review]

**Pull request merged**